woltka/classify.py

```diff
@@ -65,8 +65,8 @@ def assign_free(subs, tree, root, subok=False):
         return None if lca == root else lca
 
 
-def assign_rank(subs, rank, tree, rankdic, root, above=False, major=None,
-                ambig=False):
+def assign_rank(subs, rank, tree, rankdic=None, root=None, above=False,
```

Comment:
Now that rankdic is optional, I am uncertain if find_rank (https://github.com/qiyunzhu/woltka/blob/master/woltka/tree.py#L483) has the capability to deal with a rankdic = None.

Reply:
@emikoifish Goo catch! You are right. `rankdic` is required for `assign_rank` otherwise it won't know which taxa belong to which rank and that will be a problem. Modified accordingly.